File: Various_Lang/binarysearch.exs
array = [1,2,3,4,5]

x = IO.gets("Enter the search term: \n")
x = String.to_integer(String.trim(x))

start = 0
last = length(array) - 1

defmodule BinarySearch do
  def binary_search(array, x, start, last) do
    if start <= last do 
        middle = div(start + last, 2)
        middle_ = Enum.at(array, middle)
        cond do
            x == middle_ ->
                middle
            x > middle_ ->
                binary_search(array, x, middle + 1, last)
            x < middle_ ->
                binary_search(array, x, start, middle - 1)
        end
    else
      -1
    end
  end
end




y = BinarySearch.binary_search(array, x, start, last)
if y != -1 do
    IO.puts("The search term was found at index #{y}")
else
    IO.puts("The search term was not found")
end

File: Various_Lang/binarySearch.rs
use std::io;

fn main() {
    let mut input = String::new();
    println!("Enter the search term: ");
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let x: i32 = input.trim().parse().expect("Invalid input");
    let y = binary_search(x);
    if y == -1{
        println!("The search term was not found");
    } else {
        println!("The search term was found at index {}", y);
    }
}

fn binary_search(x: i32) -> i32 {
    let array: [i32; 5] = [1,2,3,4,5];
    let mut start: i32 = 0;
    let mut end: i32 = array.len() as i32 - 1;
    let mut middle: i32;
    while start <= end {
        middle = (start + end) / 2;
        if x == array[middle as usize]{
            return middle;
        } else if x > array[middle as usize] {
            start = middle + 1;
        } else if x < array[middle as usize] {
            end = middle - 1;
        }
    }
    return -1;
}
